proposals/CSP.md

```diff
@@ -49,6 +49,83 @@ CSP turns the problem around. Assuming unrestricted capabilities, what code is
 the developer willing to run on their site? Thus for WebAssembly, CSP will be
 used to define what sources for Wasm bytes are trusted to instantiate and run.
 
+### Summary of WebAssembly APIs and Their Risks
+
+This section introduces the APIs provided by WebAssembly that are relevant to
+Content Security Policy.
+
+Executing WebAssembly has several steps. First there are the raw WebAssembly
+bytes, which typically are loaded using the [fetch
+API](https://fetch.spec.whatwg.org/). Next, the bytes are compiled using
+`WebAssembly.compile` or `WebAssembly.compileStreaming` into a
+`WebAssembly.Module`. This module is not yet executable, but WebAssembly
+implementations may choose to translate the WebAssembly code into machine code
+at this step (Chrome does this, WebKit does not). Finally, a WebAssembly module
+is combined with an _import object_ using `WebAssembly.instantiate` to create an
+`WebAssembly.Instance` object. The import object, broadly, defines the
+capabilities of the resulting instance, oprtionally including a
+`WebAssembly.Memory`, bindings for the WebAssembly function imports, and an
+indirect function call table.  It is at this point that the WebAssembly code is
+actually executable, as the host code can call WebAssembly functions through the
+instance's exports.
+
+These steps provide the core of the WebAssembly API, but there are several other
+methods provided as well. These are summarized below along with their risks that
+are related to CSP.
+
+[**`WebAssembly.validate`**](https://webassembly.github.io/spec/js-api/index.html#dom-webassembly-validate)
+checks whether the given bytes comprise a valid WebAssembly program. In other
+words, it checks whether the bytes are syntactically correct and valid according
+to the WebAssembly type system.
+
+_Risks:_ None.
+
+[**`new WebAssembly.Module`**](https://webassembly.github.io/spec/js-api/index.html#dom-module-module)
+synchronously creates a `WebAssembly.Module` from WebAssembly bytes. This is a
+synchronous version of `WebAssembly.compile`.
+
+_Risks:_ many implementations will generate machine code at this step, even
+though it is not yet exposed as executable code to the surrounding program. It's
+possible that this code be used to build an exploit by taking advantage of
+another bug in the implementation.
```

Comment:
I thought your previous PR made this risk explicitly out of scope?

Reply:
Yes, but I decided to list it here because it has come up in discussion. Further down I reaffirm that the risks with `compile*` are out of scope.

Reply:
I'd reaffirm it here instead.